### PR TITLE
New API site redirect

### PR DESCRIPTION
**Pull request by lluiscampos**
Hello,

Two commits with two different review audience. See below:

* MEN-3825: Remove APIs directory and add API redirect instead
From the next release on, the content will not be added here by the
build script but just build independently and served from a separate
image.
@michaelatmender can you review this?

* MEN-3825: Update all links pointing to API sections
These now follow a completely different URL schema (single page,
endpoints as anchors) and we don't have any more OS/Enterprise split.
Modified checklinks white list and documentation accordingly.
@mirzak @drewmoseley @oleorhagen can you guys review this one? I don't need the formal review of all of you, but I think it is important to be aware of this change.

I tested the links locally, and they work (which is kind of surprising!). For now you just need to trust me, as there is no easy way to bring up all the setup in their different feature branches...

Also, this will lead to conflicts updating `hosted` branch later, but I will deal with that don't worry.

### CODE REVIEW

File: 08.Server-integration/01.Using-the-apis/docs.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ taxonomy:
     category: docs
 ---
 
-The Mender server microservices are all accessible using an HTTPS API. These APIs can be used to configure the server (for example, preauthorizing devices) or implementing custom workflows (for example, integrating the Mender server into an existing device management system.) There are separate APIs for [Open Source](../../200.APIs/01.Open-source/docs.md) and [Enterprise](../../200.APIs/02.Enterprise/docs.md).
+The Mender server microservices are all accessible using an HTTPS API. These APIs can be used to configure the server (for example, preauthorizing devices) or implementing custom workflows (for example, integrating the Mender server into an existing device management system.) The APIs are documented in [API chatper](../../200.API).
```
Comment (michaelatmender) [inline 08.Server-integration/01.Using-the-apis/docs.md:7]: Typo on "chapter"

File: 08.Server-integration/02.Preauthorizing-devices/docs.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ keys-client-generated/
 
 ## Preauthorize your device
 
-Now that we have the device's identity and public key, we will use the Mender server management REST APIs to preauthorize it. The APIs are documented for both [Open Source](../../200.APIs/01.Open-source/02.Management-APIs/docs.md) and [Enterprise](../../200.APIs/02.Enterprise/02.Management-APIs/docs.md).
+Now that we have the device's identity and public key, we will use the Mender server management REST APIs to preauthorize it. The APIs are documented in [API chatper](../../200.API/#management-apis).
```
Comment (michaelatmender) [inline 08.Server-integration/02.Preauthorizing-devices/docs.md:95]: "chapter"

File: 200.API/chapter.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+title: API
+redirect: https://docs.mender.io/api/
```
Comment (michaelatmender) [inline 200.API/chapter.md:3]: Depending on the branch, this should also have the version appended to the url as well.
So for 2.5 branch for example, redirect should be https://docs.mender.io/api/2.5. If this is for master branch, it should maybe have "development" on the url

File: 08.Server-integration/01.Using-the-apis/docs.md (hunk 1 above)
Comment (lluiscampos) [reply 08.Server-integration/01.Using-the-apis/docs.md:7]: Thanks. I copy-pasted this sentence, so the typo is everywhere :)

File: 200.API/chapter.md (hunk 3 above)
Comment (lluiscampos) [reply 200.API/chapter.md:3]: Yes, that is correct. I am working on a modification for the `release_tool` for this link.

**Pull request merged**